MarvelCharsApp/MarvelCharsApp/ViewModels/CategoriesViewModel.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  CategoriesViewModel.swift
+//  MarvelCharsApp
+//
+//  Created by User on 29/11/21.
+//
+
+import Foundation
+
+protocol CategoriesViewModelDelegate {
+    func didFetchSuccessfuly(categories: [CategoryModel])
+    func didFailFetching(error: Error)
+}
+
+class CategoriesViewModel {
```

Comment:
Instead of being called categoriesViewModel, it should be called like the view it uses. as it can be confused with another view name.
For example if its used on SideMenuViewController only, it should be SideMenuViewModel

Reply:
but if its used on another viewcontroller also, it could be named something generic like this.